justfile

```
@@ -11,16 +11,36 @@ default:
 zkevm-node:
     cargo run --all-features --bin zkevm-node
 
-demo *args:
-   scripts/check-architecture
-   {{compose}} up --wait {{args}}
+demo *args: (demo-profiles "zkevm1" "zkevm1-preconfirmations" "zkevm2" "zkevm2-preconfirmations" args)
 
-down:
-   {{compose}} down
+demo-profiles *args:
+    scripts/check-architecture
+
+    # When we build the L1 image locally, we create an env file with information about the
+    # deployment. However, if we are using the image pulled from the container registry, this
+    # generated file may not exist. For the following command we need to ensure the file exists, but
+    # it is ok if it is empty -- we will just use the defaults from .env.
+    touch .env.geth
+
+    # The files .env and .env.geth both contain important information (.env for defaults, .env.geth
+    # for environment variables that change when we rebuild the Geth image, such as block period and
+    # contract addresses). Newer versions of docker-compose (>= 2.17) allow you to override select
+    # variables from .env using `--env-file <another-env>`, but older versions ignore all but the
+    # last file passed with `--env-file`. Docker Desktop for Mac comes with docker-compose 2.2, and
+    # there doesn't seem to be an obvious way to update it. So for convenience, we support older
+    # versions by manually sourcing .env.geth, overriding any variables which were also set in .env.
+    scripts/source-dotenv .env.geth scripts/demo-with-profiles {{args}}
+
+down *args:
+   {{compose}} down {{args}}
 
 pull:
     {{compose-anvil}} pull && {{compose}} pull
 
+    # We are now using the default Geth image, so clear information generated when we built a local
+    # version of the image.
+    rm .env.geth
```

```suggestion
    rm -f .env.geth
```
To avoid failure if file doesn't exist.